src/ngraph/runtime/hybrid/hybrid_backend.cpp

```diff
@@ -116,70 +97,106 @@ bool runtime::hybrid::HybridBackend::call(shared_ptr<Function> func,
     // Get FunctionInstance
     bool rc = true;
 
-    auto it = m_function_map.find(func);
-    if (it == m_function_map.end())
+    using node_map_t = unordered_map<shared_ptr<Node>, shared_ptr<runtime::Tensor>>;
+
+    auto fit = m_function_map.find(func);
+    if (fit == m_function_map.end())
     {
         throw runtime_error("compile() must be called before call().");
     }
-    FunctionInstance& instance = it->second;
+    FunctionInstance& instance = fit->second;
 
     // Parameter and result node in sub_function maps to one Tensor
-    unordered_map<shared_ptr<Node>, shared_ptr<runtime::Tensor>> map_node_to_tensor_view;
+    node_map_t map_node_to_tensor;
     for (size_t i = 0; i < inputs.size(); ++i)
     {
-        map_node_to_tensor_view[instance.m_function->get_parameters()[i]] = inputs[i];
+        map_node_to_tensor[instance.m_function->get_parameters()[i]] = inputs[i];
     }
     for (size_t i = 0; i < outputs.size(); ++i)
     {
-        map_node_to_tensor_view[instance.m_function->get_results()[i]] = outputs[i];
+        map_node_to_tensor[instance.m_function->get_results()[i]] = outputs[i];
     }
 
     // Call subfunctions
-    for (shared_ptr<Function>& sub_function : instance.m_sub_functions)
+    for (const shared_ptr<Function>& sub_function : instance.m_sub_functions)
     {
         // Init backend
-        size_t placement = get_colocated_function_placement_size(sub_function);
-        auto backend = m_backend_list[placement].second;
+        size_t placement = runtime::hybrid::get_colocated_function_placement(sub_function);
+        auto backend = m_backend_list[placement];
 
-        // Prepare parameter TensorViews
-        vector<shared_ptr<runtime::Tensor>> parameter_tvs;
-        for (auto parameter_node : sub_function->get_parameters())
+        // Prepare parameter Tensors
+        vector<shared_ptr<runtime::Tensor>> parameters;
+        for (const shared_ptr<op::Parameter>& parameter_node : sub_function->get_parameters())
         {
-            if (map_node_to_tensor_view.find(parameter_node) != map_node_to_tensor_view.end())
+            auto it = map_node_to_tensor.find(parameter_node);
+            if (it != map_node_to_tensor.end())
             {
-                parameter_tvs.push_back(map_node_to_tensor_view.at(parameter_node));
+                if (it->second->get_parent() == backend.get())
+                {
+                    parameters.push_back(it->second);
+                }
+                else
+                {
+                    auto parameter = backend->create_tensor(parameter_node->get_element_type(),
+                                                            parameter_node->get_shape());
+                    parameter->copy_from(*(it->second));
+                    parameters.push_back(parameter);
```

Comment:
this logic is very similar to results' one? I was wondering if we could maybe refactor it into a helper?

Reply:
There is a lot of rework to go on here but that really belongs in a separate PR.